### PR TITLE
Reorg 2017 Routing

### PR DESCRIPTION
**Pull request by sonukapoor**
Connects to #258 #259 #260 #261 #262 #263 #264 #265 #280

### CODE REVIEW

**Comment by sonukapoor** [comment]
@gvwilson @rpdasilva I am going to add the other FIXME issues of the routing section into this PR. Please hold off until I am done with that before reviewing.

**Comment by sonukapoor** [comment]
Ready for review @gvwilson @rpdasilva @barretodavid

File: slides/src/content/introduction.md [hunk 1]
```diff
@@ -37,4 +53,247 @@ FIXME
...
+
+1. Install NVM (Node Version Manager) -- [nvm.sh](http://nvm.sh)
+1. Install the latest LTS version of node -- `nvm install --lts`
+1. Install Angular CLI -- `npm install --global @angular/cli`
+
+---
+
+## Creating a New Project
+
+`ng create [project-name]`
```
Comment (rpdasilva) [inline slides/src/content/introduction.md]: Should be `ng new [project-name]`

File: slides/src/content/routing.md [hunk 2]
```diff
@@ -114,12 +113,64 @@ For example the following code defines a link to the route at path `component-on
...
+
+---
+
+## Defining Links Between Routes (2/2)
+
+The path array can be seen as segments of an URL. 
+For example, using the following path in our array:
+
+```javascript
+var path = ['component-one', 'param1', 'param2'];
```
Comment (rpdasilva) [inline slides/src/content/routing.md]: In some cases we're using a preceding `/` with our path segments, sometimes not - either make this consistent, or add a few words explaining what `/` means

File: slides/src/content/routing.md [hunk 3]
```diff
@@ -169,27 +220,100 @@ The reason that the `params` property on `ActivatedRoute` is an Observable is th
...
+---
 
 ## Route Authorization (1/3)
 
 To control whether the user can navigate to or away from a given route, we can use route guards.
 
 In order to use route guards, we must register them with the specific routes we want them to run for.
 
-```
+```javascript
```
Comment (rpdasilva) [inline slides/src/content/routing.md]: Should be `ts`

File: slides/src/content/routing.md [hunk 4]
```diff
@@ -231,60 +355,115 @@ The canDeactivate function passes the component being deactivated as an argument
...
+    ]
+```
+
+This way we can get rid of the first empty item, which essentially just redirects the user to the `overview` route.
+
+---
+
+## Child Routes (4/4)
+
+To display the `overview` or `specs` section of a parent, the child route will need the product ID from the parent. The child route component can access the parent route's parameters as follows using the `route` and `activatedRoute`
```
Comment (rpdasilva) [inline slides/src/content/routing.md]: `ActivatedRoute`

File: slides/src/content/routing.md [hunk 5]
```diff
@@ -231,60 +355,115 @@ The canDeactivate function passes the component being deactivated as an argument
...
+    this.sub.unsubscribe();
+  }
 }
 ```
 
-[View Example](http://plnkr.co/edit/Ko6VFRGRmu5jJ9ArwxvC?p=preview)
-
 ---
+
+FIXME: What are child routes
```
Comment (rpdasilva) [inline slides/src/content/routing.md]: Can be removed

File: slides/src/index.html [hunk 6]
```diff
@@ -17,6 +17,19 @@
...
       </section>
 
+      <section
+        data-markdown="content/routing.md"
+        data-separator="^(\r?)\n---(\r?)\n"
+        data-separator-vertical="^(\r?)\n\+\+\+(\r?)\n"
+        data-separator-notes="^Notes:">
+      </section>
+
+      <section
```
Comment (rpdasilva) [inline slides/src/index.html]: Remove

**Pull request merged**